polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -356,6 +354,69 @@ ValueCategory MLIRScanner::VisitPredefinedExpr(clang::PredefinedExpr *Expr) {
 }
 
 ValueCategory MLIRScanner::VisitInitListExpr(clang::InitListExpr *Expr) {
+  LLVM_DEBUG({
+    llvm::dbgs() << "VisitInitListExpr: ";
+    Expr->dump();
+    llvm::dbgs() << "\n";
+  });
+
+  assert(!Expr->hadArrayRangeDesignator() && "Unsupported");
+
+  const auto Loc = getMLIRLocation(Expr->getExprLoc());
+
+  if (auto VType = Glob.getTypes()
+                       .getMLIRType(Expr->getType())
+                       .dyn_cast<mlir::VectorType>()) {
+    int64_t ResElts = VType.getNumElements();
+
+    int64_t CurIdx = 0;
+    auto V = ValueCategory::getNullValue(Builder, Loc, VType);
+    for (auto *IE : Expr->children()) {
+      ValueCategory Init = Visit(IE);
+      SmallVector<int64_t> Args;
+
+      auto VVT = Init.val.getType().dyn_cast<mlir::VectorType>();
+
+      // Handle scalar elements.
+      if (!VVT) {
+        V = V.Insert(Builder, Loc, Init.val, CurIdx);
+        ++CurIdx;
+        continue;
+      }
+
+      int64_t InitElts = VVT.getNumElements();
+
+      int64_t Offset = (CurIdx == 0) ? 0 : ResElts;
+
+      // Extend init to result vector length, and then shuffle its contribution
+      // to the vector initializer into V.
+      if (Args.empty()) {
```

Comment:
Why do we need to check if Args is empty? it is created at line 376 and never changed till here.